### PR TITLE
feat: companycardgrid component

### PR DESCRIPTION
**Pull request by ranerhei**
Made a Grid component that takes in a array of logos / websitelinks to make CompanyCards

### CODE REVIEW

File: components/company/Card.vue [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+      max-width="250"
+      v-bind="props"
+      @click="redirectToWebsite"
+    >
+      <v-img :width="500" aspect-ratio="4/3" cover :src="content.logo"></v-img>
+    </v-card>
+  </v-hover>
+</template>
+
+<script setup lang="ts">
```
Comment (AasmundN) [inline components/company/Card.vue:17]: Place script at top of file

File: components/company/Card.vue [hunk 2]
```diff
@@ -0,0 +1,30 @@
+<template>
+  <v-hover v-slot="{ isHovering, props }">
```
Comment (AasmundN) [inline components/company/Card.vue:2]: Use PascalCase for all components

File: components/company/Grid.Vue
Comment (ipeglin) [inline components/company/Grid.Vue:1]: Here we should either pass full `CompanyCard` components as props, or as children.

Examples of the different approaches are as follows.

## _`CompanyCard` as props array:_
```vue
<!--ParentComponent.vue -->
<script setup lang="ts">
  const companies= ref([
    { uid: 1, name: 'Company 1' },
    { uid: 2, name: 'Company 2' },
    // ... other companies
  ]);
</script>

<template>
  <CompanyGrid :content="companies" />
</template>
```
```vue
<!-- CompanyGrid.vue -->
<script setup>
  defineProps({
  content: { type: Array as PropType<CompanyCard[]>, required}, // If I'm not mistaken
  })
</script>

<template>
  <div class="card-grid">
    <CompanyCard v-for="company in companies" :key="company.uid" :company="company" />
  </div>
</template>
```

## Or, _`CompanyCard` as children:_
```vue
<!-- CompanyGrid.vue -->
<template>
  <div class="card-grid">
    <slot></slot>
  </div>
</template>
```

```vue
<!-- ParentComponent.vue -->
<script setup lang="ts">
  const companies= ref([
    { uid: 1, name: 'Company 1' },
    { uid: 2, name: 'Company 2' },
    // ... other companies
  ]);
</script>

<template>
  <CompanyGrid>
    <CompanyCard v-for="company in companies" :key="company.uid" :company="company" />
  </CompnanyGrid>
</template>
```

File: pages/index.vue [hunk 3]
```diff
@@ -14,6 +14,13 @@
       "
       >Create user</VBtn
     >
+    <CompanyCard
```
Comment (ipeglin) [inline pages/index.vue:17]: CompanyCard should be passed as child or props. See the above comment attached to this file

**Review by ipeglin — CHANGES_REQUESTED** [review]
Could you have a look at how we pass information to the Grid component? You don't necessarily have to pass the whole `Company` objects as we only need `logo` and `website`. However, I think this should be handled in the `CompanyCard` component, and not `CompanyGrid`. This way, we can simply pass complete `CompanyCard`s to the Grid component, without using computations in `CompanyGrid` as a middleman to creating `CompanyCards`. Don't know if my comments are clear enough to understand, but we can go through it on tuesday :) If I have missunderstood, please say so too. @ranerhei @AasmundN

File: components/company/Card.vue [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    logo: string
+    website: string
+  }
+
+  const props = defineProps({
+    content: { type: Object as PropType<CompanyCardContent>, required: true },
+  })
+
+  const redirectToWebsite = () => {
+    window.open(props.content.website)
```
[the remark this answers is not in the document]
Comment (ranerhei) [reply components/company/Card.vue]: fixed in another pr

File: components/company/Card.vue (hunk 1 above)
Comment (ranerhei) [reply components/company/Card.vue]: fixed

File: components/company/Grid.Vue [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+  defineProps({
+    content: { type: Array as PropType<CompanyInformation[]>, required: true },
+  })
+</script>
+
+<template>
+  <v-container>
+    <v-row>
+    <!-- Style styrer høyde og slikt-->
+      <v-col v-for="n in content">
```
[the remark this answers is not in the document]
Comment (ranerhei) [reply components/company/Grid.Vue]: should be fixed

File: components/company/Grid.Vue [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+    website: string
+  }
+
+  defineProps({
+    content: { type: Array as PropType<CompanyInformation[]>, required: true },
+  })
+</script>
+
+<template>
+  <v-container>
```
[the remark this answers is not in the document]
Comment (ranerhei) [reply components/company/Grid.Vue]: should be fixed

File: components/company/Grid.Vue [hunk 7]
```diff
@@ -0,0 +1,28 @@
+<script setup lang="ts">
+  import { PropType } from 'nuxt/dist/app/compat/capi'
```
[the remark this answers is not in the document]
Comment (ranerhei) [reply components/company/Grid.Vue]: should be removed now

File: components/company/Card.vue
[the remark this answers is not in the document]
Comment (ranerhei) [reply components/company/Card.vue:1]: Think this is fixed

File: pages/index.vue [hunk 8]
```diff
@@ -27,5 +27,15 @@ appspot.com/storage_companies%2FSony%2FLogo%2Fsony.png`,
       }"
       class="my-5 d-flex justify-center"
     />
+
+    <CompanyGrid>
+      <CompanyCard
+        :content="{
+          logo: `https://1000logos.net/wp-content/
+          uploads/2021/05/Sony-logo.png`,
```
Comment (AasmundN) [inline pages/index.vue:35]: Put this on one line, image won't render otherwise.

**Review by AasmundN — DISMISSED** [review]
Looks good

File: pages/index.vue (hunk 8 above)
Comment (ranerhei) [reply pages/index.vue:35]: this is just a test case, not the real code? was thinking of removing it before merging

Comment (ranerhei) [reply pages/index.vue:36]: actually, I can't put it on a single line, get a lint error. CompanyHero has the same solution as me, and it renders fine. Tested

**Review by AasmundN — APPROVED** [review]

**Pull request merged**